cylc/flow/network/resolvers.py

```diff
@@ -329,14 +329,33 @@ async def get_workflows(self, args):
             args)
 
     # nodes
+    def get_node_state(self, node, node_type):
+        """Return node state, from node or data."""
+        if node_type in DEF_TYPES:
+            return None
+        # Don't lookup data-store state in getattr, as it might not be there.
+        # (args are evaluated first)
+        state = getattr(node, 'state', None)
+        # If node has no state, it must be in the data-store.
+        if not state:
+            return self.data_store_mgr.data[
+                Tokens(node.id).workflow_id
+            ][node_type][node.id].state
+        return state
```

Comment:
```suggestion
        # If node has no state, it must be in the data-store.
        return (
            getattr(node, 'state', None) or
            self.data_store_mgr.data[
                Tokens(node.id).workflow_id
            ][node_type][node.id].state
        )
```

Reply:
(No need for the intermediate variable).

Reply:
Fixed.